core/core.php

```diff
@@ -153,21 +145,23 @@ public function get_ext_img_path()
 	}
 
 	/**
-	 * Validates the bade order an fixes it if necessary.
+	 * Validates the bade order and fixes it if necessary.
 	 * Returns true on success.
 	 *
 	 * @return bool
 	 */
 	public function validate_badge_order($user_id)
 	{
 		$sql = 'SELECT item_id, badge_order
-		FROM ' . $this->table_prefix . 'openra_user_badges' . '
-		WHERE user_id = ' . $user_id . '
-		ORDER BY badge_order';
+			FROM ' . $this->table_prefix . 'openra_user_badges' . '
+			WHERE user_id = ' . $user_id . '
```

Comment:
Can we please cast `$user_id` to int here to guarantee safety?

Reply:
I moved the query to the module, it is only used in one place anyway.